source/ed/connectors/fusio_parser.cpp

```diff
@@ -1266,6 +1269,57 @@ void CommentLinksFusioHandler::handle_line(Data&, const csv_row& row, bool) {
     }
 }
 
+void ObjectCodesFusioHandler::init(Data&){
+    object_uri_c = csv.get_pos_col("object_id");
+    object_type_c = csv.get_pos_col("object_type");
+    code_c = csv.get_pos_col("object_code");
+    object_system_c = csv.get_pos_col("object_system");
+}
+
+static nt::Type_e type_by_caption(std::string& caption){
+    boost::algorithm::to_lower(caption);
+    if (caption == "line")
+        return nt::Type_e::Line;
+    if (caption == "route")
+        return nt::Type_e::Route;
+    if (caption == "network")
+        return nt::Type_e::Network;
+    if (caption == "trip")
+        return nt::Type_e::VehicleJourney;
+    if (caption == "stop_area")
+        return nt::Type_e::StopArea;
+    if (caption == "stop_point")
+        return nt::Type_e::StopPoint;
+    return nt::Type_e::Unknown;
+}
+
+void ObjectCodesFusioHandler::handle_line(Data& data, const csv_row& row, bool) {
+    if(! has_col(object_uri_c, row) || ! has_col(object_type_c, row) || ! has_col(code_c, row) || ! has_col(object_system_c, row)) {
+        LOG4CPLUS_FATAL(logger, "Error while reading " + csv.filename +
+                        "  impossible to find all needed fields");
+        throw InvalidHeaders(csv.filename);
+    }
+
+    ed::types::ObjectCode* object_code = new ed::types::ObjectCode();
```

Comment:
I think there is a leak there. I think the best way to avoid that is to do a `ed::types::ObjectCode` instead of a `ed::types::ObjectCode`